x-pack/plugins/apm/public/context/UrlParamsContext/constants.ts

```diff
@@ -4,11 +4,72 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import { IUiSettingsClient } from 'kibana/public';
+import { UI_SETTINGS } from '../../../../../../src/plugins/data/common';
+
 export const TIME_RANGE_REFRESH = 'TIME_RANGE_REFRESH';
 export const LOCATION_UPDATE = 'LOCATION_UPDATE';
-export const TIMEPICKER_DEFAULTS = {
+
+interface TimePickerQuickRange {
+  from: string;
+  to: string;
+  display: string;
+}
+
+interface ITimePickerDefault {
+  rangeFrom: string;
+  rangeTo: string;
+  refreshPaused: boolean;
+  refreshInterval: number;
+  quickRanges: TimePickerQuickRange[];
+}
+
+interface TimePickerRefreshInterval {
+  pause: boolean;
+  value: number;
+}
+
+interface TimePickerTimeDefaults {
+  from: string;
+  to: string;
+}
+
+export let timePickerDefault: ITimePickerDefault = {
   rangeFrom: 'now-24h',
   rangeTo: 'now',
-  refreshPaused: 'true',
-  refreshInterval: '0',
+  refreshPaused: true,
+  refreshInterval: 10000,
+  quickRanges: [],
 };
+
+export function getTimePickerQueryParam() {
```

Comment:
Is it necessary to have a getter when `timePickerDefault` is already exported? And why "queryParam" in the name `getTimePickerQueryParam`?

Reply:
Because `getTimePickerQueryParam` will be used to set the query params in the URL.